Introduce OSI Traffic Participants

pmai opened the pull request:
This PR is part of the SETLevel4To5 extensions of OSI and OSMP, introducing Traffic Participant as a model kind, as well as optionally separating out static content handling for planning and efficiency purposes:

* A new model kind of Traffic Participant is added, which has one or more SensorView inputs, an optional TrafficCommand input, and a TrafficUpdate output which is fed back into the simulation. The optional TrafficCommand input allows control of the traffic participant by e.g. a scenario engine.

* Optional GroundTruthInit parameters are introduced to all model kinds to allow receiving a complete set of static ground truth data prior to initialization start; this enables inter alia complex pre-processing to occur without endangering real-time performance.

   As a corollary a PR to add a flag to omit static ground truth content from SensorView to the SensorViewConfiguration message is created in OSI. In combination this approach allows the reduction of transferred data by eliminating redundant transfer of static content.

   It is currently on purpose left undefined, what exactly constitutes static content, since this definition should be done across OSI as a whole through the ASAM development process (i.e. to answer questions like: Are StationaryObjects static, even if they could be moved - e.g. traffic cones?, etc.).

The setup described by these changes is being used as mainbranch development platform for SETLevel4To5 model development.

It is being provided as a draft PR to aid in discussion of those changes in the relevant ASAM working packages. Additional changes from SETLevel4To5 might be pushed to the branch when released.

#### Check the checklist

- [X] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation) for osi-sensor-model-packaging.
- [X] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [X] New and existing unit tests / travis ci pass locally with my changes.

pmai committed:
Add traffic participant models as model type

Signed-off-by: Pierre R. Mai <pmai@pmsf.de>

tbleher committed:
Add OSI Traffic Participant Initialization to the spec

Signed-off-by: Pierre R. Mai <pmai@pmsf.de>

tbleher committed:
Relax requirements on traffic participant initialization

A few of the stated requirements were not actually needed. Relax these,
to make it simpler for simulation environments to provide the needed
data to the models.

Signed-off-by: Pierre R. Mai <pmai@pmsf.de>

tbleher committed:
Traffic Participants: Make OSMPGroundTruthInit optional

In case a traffic participant does not need the initialization message, it
can be omitted.

Signed-off-by: Pierre R. Mai <pmai@pmsf.de>

pmai committed:
Adjust wording on potential dynamic content of init

Signed-off-by: Pierre R. Mai <pmai@pmsf.de>

pmai committed:
Generalize GroundTruth init parameter across models

Signed-off-by: Pierre R. Mai <pmai@pmsf.de>

kmeids commented:
**Output from CCB meeting - 11.11.2020**

Actions:
1. Documentation review to be performed before the minor release.
2. Pull request to be reviewed by @clemenshabedank 
3. Pull request to be merged by @pmai **after** documentation review.

clemenshabedank committed:
Minor consistency changes for traffic participants

Fix missing mention of traffic participant models in first sentence.
Use version=3.3.0 for examples where top-level messages are introduced
in V3.3.

Signed-off-by: Pierre R. Mai <pmai@pmsf.de>

clemenshabedank commented:
@pmai will we tag this version as 1.1.0? My changes then have a typo (i typed 1.2.0). Other than that it would be ready to merge in my opinion.

pmai commented:
@clemenshabedank yes, 1.1.0 would be better. Will change while rebasing for merge. Thanks for the clarifications!
